tests/ops/op_math/test_controlled_ops.py

```diff
@@ -428,3 +441,140 @@ def test_unitary_check(self):
+        assert np.allclose(op.pow(n)[0].matrix(), op.matrix())
+        assert op.pow(n)[0].name == op.name
+
+    @pytest.mark.parametrize("op", period_two_ops)
+    @pytest.mark.parametrize("n", (2, 6, 0, -2))
+    def test_period_two_pow_even(self, op, n):
+        """Test that ops with a period of 2 raised to an even power are empty lists."""
+        assert len(op.pow(n)) == 0
+
+    @pytest.mark.parametrize("op", period_two_ops)
+    def test_period_two_noninteger_power(self, op):
+        """Test that ops with a period of 2 raised to a non-integer power raise an error."""
+        if op.__class__ is qml.CZ:
+            pytest.skip("CZ can be raised to any power.")
+        with pytest.raises(qml.operation.PowUndefinedError):
+            op.pow(1.234)
+
+    @pytest.mark.parametrize("n", (0.12, -3.462, 3.693))
+    def test_cz_general_power(self, n):
+        """Check that CZ raised to an non-integer power that's not the square root
+        results in a controlled PhaseShift."""
+        op_pow = qml.CZ(wires=[0, 1]).pow(n)
+
+        assert len(op_pow) == 1
+        assert isinstance(op_pow[0], qml.ops.ControlledOp)
+        assert isinstance(op_pow[0].base, qml.PhaseShift)
+        assert qml.math.allclose(op_pow[0].data[0], np.pi * (n % 2))
+
+
+class TestControlledMethod:  # pylint: disable=too-few-public-methods
+    """Tests for the _controlled method of non-parametric operations."""
+
+    def test_CZ(self):
+        """Test the PauliZ _controlled method."""
+        out = qml.CZ(wires=[0, 1])._controlled("a")  # pylint: disable=protected-access
+        assert qml.equal(out, qml.CCZ(("a", 0, 1)))
+
+
+class TestSparseMatrix:  # pylint: disable=too-few-public-methods
+    @pytest.mark.parametrize("op, mat", SPARSE_MATRIX_SUPPORTED_OPERATIONS)
+    def test_sparse_matrix(self, op, mat):
+        expected_sparse_mat = csr_matrix(mat)
+        sparse_mat = op.sparse_matrix()
+
+        assert isinstance(sparse_mat, csr_matrix)
+        assert isinstance(expected_sparse_mat, csr_matrix)
+        assert all(sparse_mat.data == expected_sparse_mat.data)
+        assert all(sparse_mat.indices == expected_sparse_mat.indices)
+
+
+label_data = [
+    (qml.CZ(wires=(0, 1)), "Z"),
+]
+
+
+@pytest.mark.parametrize("op, label", label_data)
+def test_label_method(op, label):
+    assert op.label() == label
+    assert op.label(decimals=2) == label
+
+
+control_data = [
+    (qml.CZ(wires=(0, 1)), Wires(0)),
+]
+
+
+@pytest.mark.parametrize("op, control_wires", control_data)
+def test_control_wires(op, control_wires):
+    """Test ``control_wires`` attribute for non-parametrized operations."""
+
+    assert op.control_wires == control_wires
+
+
+involution_ops = [  # ops who are their own inverses
+    qml.CZ((0, 1)),
+]
+
+
+@pytest.mark.parametrize("op", involution_ops)
+def test_adjoint_method(op):
+    adj_op = copy.copy(op)
+    for _ in range(4):
+        adj_op = adj_op.adjoint()
+
+        assert adj_op.name == op.name
```

Comment:
```suggestion
        assert qml.equal(adj_op, op)
```
You would want this test to fail if `op` was something like `CRX`.  The name would still be the same, but the parameter would be different.  This test wouldn't catch that.